### DISCUSSION TITLE
Is there any way to create dag realtime?

### DISCUSSION THREAD

**Discussion by yuhua6380** [apache/airflow#15421] [General] [answered]
Is there anyway to create a new dag realtime? At least can load or refresh specific dag throught api or restfulAPI.
In my case, i want to create a dynamic dag and run it as soon as possiable.

**Comment by turbaszek** [comment, marked as answer]
DAG is a file. So only way to create a new DAG is to upload a file. If you know what logic the DAG  should execute you can try to utilise the dag configuration as an input for adjusting this logic.

**Reply by yuhua6380** [reply]
DAG configuration can solve the dynamic problem. But there is still a problem. It's the consistence API. For example, how can I run the DAG as soon as I uploaded the DAG without consistence problems?

**Reply by RafayAK** [reply]
Check out this project: https://github.com/ajbosco/dag-factory
